react/src/components/ManageAppsModal.tsx

```diff
@@ -1,101 +1,256 @@
 import BAIModal, { BAIModalProps } from './BAIModal';
 import { useWebComponentInfo } from './DefaultProviders';
 import Flex from './Flex';
-import { DeleteOutlined, PlusOutlined, CheckOutlined } from '@ant-design/icons';
-import { Input, Typography, Space, Button, Form } from 'antd';
-import React from 'react';
-import { useTranslation } from 'react-i18next';
+import {
+  KVPairInput,
+  ManageAppsModalMutation,
+} from './__generated__/ManageAppsModalMutation.graphql';
+import { DeleteOutlined, PlusOutlined } from '@ant-design/icons';
+import { Input, Button, Form, message, Typography, App } from 'antd';
+import graphql from 'babel-plugin-relay/macro';
+import React, { useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { useMutation } from 'react-relay';
 
-interface Props extends BAIModalProps {}
-const { Text } = Typography;
-const ManageAppsModal: React.FC<Props> = ({ ...baiModalProps }) => {
+const ManageAppsModal: React.FC<BAIModalProps> = ({ ...baiModalProps }) => {
   const { t } = useTranslation();
+  const [form] = Form.useForm();
+  const app = App.useApp();
+
+  const [open, setOpen] = useState<boolean>(true);
+  const [validateDetail, setValidateDetail] = useState<string>('');
   const { value, dispatchEvent } = useWebComponentInfo();
+
+  const [commitModifyImageInput, isInFlightModifyImageInput] =
+    useMutation<ManageAppsModalMutation>(graphql`
+      mutation ManageAppsModalMutation(
+        $target: String!
+        $architecture: String
+        $props: ModifyImageInput!
+      ) {
+        modify_image(
+          target: $target
+          architecture: $architecture
+          props: $props
+        ) {
+          ok
+          msg
+        }
+      }
+    `);
+
   let parsedValue: {
-    open: boolean;
+    image: any;
     servicePorts: any;
   };
   try {
     parsedValue = JSON.parse(value || '');
   } catch (error) {
     parsedValue = {
-      open: false,
+      image: {},
       servicePorts: [],
     };
   }
-  const { open, servicePorts } = parsedValue;
-  const onFinish = (values: any) => {
-    console.log('Saved settings. ', values);
+  const { image, servicePorts } = parsedValue;
+
+  const handleOnclick = async () => {
+    try {
+      await form.validateFields();
+
+      const values = form
+        .getFieldValue('apps')
+        .map((item: { app: string; protocol: string; port: number }) => {
+          return `${item.app}:${item.protocol}:${item.port}`;
+        })
+        .join(',');
+
+      const INPUT: KVPairInput[] = [];
+      Object.entries(image.labels).forEach(([key, value]) => {
+        if (key.includes('service-ports')) {
+          INPUT.push({ key: key, value: values });
+        } else {
+          INPUT.push({ key: key, value: value?.toString() || '' });
+        }
+      });
```

Comment:
How about using lodash?

```suggestion
const INPUT = _.map(image.labels, (value, key) => {
  if (key.includes('service-ports')) {
    return { key, value: values };
  } else {
    return { key, value: _.toString(value) || '' };
  }
});
```